### ISSUE TITLE
Missing wrapper function

### ISSUE THREAD

**Issue by Almaju** [floydspace/serverless-esbuild#157]
Hello,

When I do `sls deploy`, the bundle is correctly built however the handler is set to `s_function.handler` for a function located in let's say `src/function.js`. It works if I manually rename it to `src/function.handler` in the AWS console but it gets overwritten after every deploy.

I can reproduce that with any of the esbuild starter I have found. Even with https://github.com/floydspace/serverless-esbuild/tree/master/examples/minimal

Note that when doing `sls package` or `sls deploy`, I can see the file `s_function.handler` being created then deleted on my root dir.

**Comment by yemi** [comment]
For some reason, this seems to happen only when I have `org` and `app` specified in `serverless.yml`, if I remove them, the `"Handler"` in the generated `cloudformation-template-update-stack.json` gets set to `myfunction.handler`.

The only difference I can see in the logs is this:
```
Serverless: Zip service my-service - 106.52 KB [54 ms]
Serverless: Packaging service...
```

vs.

```
Serverless: Zip service my-service - 106.52 KB [57 ms]
Serverless: Packaging service...
Serverless: Installing dependencies for custom CloudFormation resources...
```

It seems that by "Installing dependencies for custom CloudFormation resources" it also somehow always sets the `"Handler"` in `cloudformation-template-update-stack.json` to `s_myFunction.handler`, which of course doesn't exist.

Any ideas?

**Comment by joewestcott** [comment]
I have the same issue. I've also got `app` and `org` set in my `serverless.yml`, so I assume we're all using the Serverless Dashboard/Framework CI/CD service.

I haven't had an issue when using serverless-webpack, so I think this plugin has an incompatibility with the serverless [dashboard-plugin](https://github.com/serverless/dashboard-plugin). My guess at this stage would be `serverless-esbuild` is setting the `Handler` attribute too late, which means the subsequent dashboard plugin is picking up an incorrect value?

⚠️   just I guess, I don't know what I'm doing...!

**Comment by joewestcott** [comment]
More digging:

It seems like `serverless-dashboard` checks for `package.artifact` of each item within the `ctx.sls.service.functions` object to perform it's function wrapping on, see [here](  https://github.com/serverless/dashboard-plugin/blob/de7ecf9d295a6f728d21b1b9d16bfbeaccb7964b/lib/wrap.js#L393).

I can see `serverless-webpack` is [setting the artifact path](https://github.com/serverless-heaven/serverless-webpack/blob/master/lib/packageModules.js#L12) by logging the state of the functions object within `before:package:createDeploymentArtifacts`, whereas I can't see the same with `serverless-esbuild`.

I'm not using `package.individually` if that makes a difference.

As far as I can tell, this is the reason causing this issue. @floydspace, any ideas, am I on the right track?

**Comment by mjyocca** [comment]
I believe I'm experiencing this same issue, mine is `s_app.handler`. Any known workarounds?

**Comment by samchungy** [comment]
Could someone send an example repo for me to clone? I see a few various setups above so I'd be interested to find the issue

**Comment by mjyocca** [comment]
@samchungy I was able to replicate this just by @joewestcott observation of adding `app` and `org` properties on the serverless config. Could prob just use this template since it has serverless-esbuild configured https://github.com/serverless/serverless/tree/master/lib/plugins/create/templates/aws-nodejs-typescript

**Comment by samchungy** [comment]
So it looks like we bundle our code before this Serverless wrapper thing overwrites the handler. It looks like there is some expectation that we bundle the extra wrapper file and serverless_sdk. I wonder how serverless-webpack works. Has anybody tried with that?

**Comment by samchungy** [comment]
So it looks like Serverless adds the dashboard plugin after our package function. The plugin registers the https://github.com/serverless/dashboard-plugin/blob/e414a135e654dade32eabc135d3d834c3ae0ee1f/lib/plugin.js#L305 hook which runs after our plugin.

No clue how to solve this. Seems like a Serverless Framework issue to me seeing as they auto inject this with no way to means to change the order

**Comment by mjyocca** [comment]
That's unfortunate, serverless-webpack works, so I wonder if serverless framework hard coded serverless-weback to run before the dashboard plugin.

**Comment by samchungy** [comment]
Interesting, I'll take a peek at webpack and see if they do something different to us

**Comment by samchungy** [comment]
Okay - so it looks like this will work when `package` `individually: true` is set. The serverless-dashboard plugin looks for artifacts set on the function level whereas when package individually is not true we use the `service.package.artifact`

**Comment by mjyocca** [comment]
Quick update, for some reason setting `package` `individually: true` the final zip is missing node_module dependencies, package.json, package-lock etc

**Comment by samchungy** [comment]
> Quick update, for some reason setting `package` `individually: true` the final zip is missing some node_module dependencies like `pg`, package.json, package-lock etc

That is likely related to https://github.com/floydspace/serverless-esbuild/pull/249. Working on it. Time is scarce for me though.

**Comment by mjyocca** [comment]
I'm going with removing the dashboard integration for now so not urgent.

**Comment by samchungy** [comment]
Closing this as this is a Serverless Dashboard Plugin issue. I've raised a PR to fix it in their repo.

https://github.com/serverless/dashboard-plugin/issues/668

**Comment by samchungy** [comment]
Hey all in here, Serverless framework people merged my fix and it's released in the latest version 3.30. if you're still after this functionality, feel free to update and try again.

https://github.com/serverless/serverless/releases/tag/v3.3.0
